rclpy/test/test_executor.py

```diff
@@ -221,6 +222,40 @@ def test_create_task_dependent_coroutines(self):
         self.assertTrue(future2.done())
         self.assertEqual('Sentinel Result 2', future2.result())
 
+    def test_create_task_during_spin(self):
+        self.assertIsNotNone(self.node.handle)
+        executor = SingleThreadedExecutor(context=self.context)
+        executor.add_node(self.node)
+
+        future = None
+
+        def spin_until_task_done(executor):
+            nonlocal future
+            while future is None or future.done():
+                try:
+                    executor.spin_once()
+                finally:
+                    executor.shutdown()
+                    break
+
+        # Start spinning in a separate thread
+        thr = threading.Thread(target=spin_until_task_done, args=(executor, ), daemon=True)
+        thr.start()
+
+        def func():
+            return 'Sentinel Result'
+
+        # Create a task
+        future = executor.create_task(func)
```

Comment:
I think there might be a race where this test could have a false negative if `spin_until_task_done` has not reached the `while` loop in `_wait_for_ready_callbacks` prior to this task being created.

Reply:
Yeah, I think you're right. I'm not sure how to avoid it, but it can be mitigated by sleeping for a short time before creating the task.

Reply:
See 1ee031a